docs(aio): add check for `ProgressEvent` inside `HttpErrorResponse` handler

maxkoretskyi opened the pull request:
According to the spec (https://xhr.spec.whatwg.org/#event-xhr-error),
XHR errors are instances of ProgressEvent,
so the existing check (e.error instance of Error) won't catch them.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?
```
[ ] Yes
[x ] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

maximusk commented:
@kara hey, can you help me figure out why the build failed?

kara commented:
@maximusk Doesn't look related to your PR. Probably part of the Travis problem yesterday, so re-running the job.

maximusk commented:
@kara , thanks, it seems to have failed again

gkalpak commented:
@maximusk, the Travis failure should be fixed by rebasing on latest master.

maximusk commented:
@gkalpak , thanks, the build has succeeded, PR can be merged

gkalpak commented on aio/content/guide/http.md:
```diff
@@ -143,7 +143,7 @@ http
     data => {...},
     (err: HttpErrorResponse) => {
-      if (err.error instanceof Error) {
+      if (err.error instanceof Error || e.error instanceof ProgressEvent) {
```
Shouldn't it be `err.error` (instead of `e.error`)?

maximusk replied:
yeah, right, udpated

maxkoretskyi committed:
docs(aio): add check for `ProgressEvent` inside `HttpErrorResponse` handler

According to the spec (https://xhr.spec.whatwg.org/#event-xhr-error),
XHR errors are instances of ProgressEvent,
so the existing check (e.error instance of Error) won't catch them.

brandonroberts commented:
Hey @maximusk! Will you rebase this branch on master and remove the scope from the git commit message?

mkoretskyi commented:
@brandonroberts , hey, do you still this one is still relevant? I'm thinking about just closing it.

brandonroberts commented:
@mkoretskyi I think it could still be useful if it helps conform to the spec but if you want to close, that's fine also.
